enumerable.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    (0...to_a.length).each do |i|
+      yield(to_a[i], i)
+    end
+    self
+  end
+
+  def my_select
+    arr = []
+    to_a.my_each do |num|
+      yield num ? arr.push(num) : ''
+    end
+    arr
+  end
+
+  def my_all?
+    to_a.my_each do |num|
+      return false unless yield num
+    end
+    true
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] when no block or argument is given returns true when none of the collection members are false or nil
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern